lte/protos/pipelined.proto

```diff
@@ -198,6 +198,217 @@ message PacketDropTableId {
+    BUFF = 2;
+    NOCP = 3;
+    DUPL = 4;
+  }
+  repeated Action apply_action = 1;
+}
+
+
+message RedirectInfo {                //8.2.20    //doubt who gives the info to SMF
+  enum RedirectAddrType {
+    IPV4 = 0;
+    IPV6 = 1;
+    URL = 2;
+    SIPURI = 3;
+    IPV4V6 = 4;
+  }
+  RedirectAddrType red_add_type = 1;
+  string red_server_addr = 2;
+}
+
+message OuterHeaderCreation {
+ uint32 o_teid = 1;
+ string ipv4_adr = 2;
+ uint32 port_no = 3;
+}
+
+
+//Table 7.5.2.3-2: Forwarding Parameters IE in FAR
+message FwdParam {
+  uint32 dest_iface = 1;                        // need to add enums for dest. interface 8.2.24
+  string net_instace = 2;
+  RedirectInfo redirect_info = 3;
+  OuterHeaderCreation outr_head_cr = 4;                  //this has to be defined as per MVC // now created
+}
+
+
+//Table 7.5.2.3-3: Duplicating Parameters IE in FAR
+message DupParam {
+  uint32 dest_iface = 1;
+}
+
+
+//PDI Message Table 7.5.2.2-2: PDI IE within PFCP Session Establishment Request
+message PDI {
+  uint32 src_interface = 1;
+  // LocalFTeid local_f_teid = 2;        //optional and conditionals
+  string net_instance = 3;
+  string ue_ip_adr = 4;         //modified and added ueipv4 address
+  bytes tr_ep_id = 5;                //8.2.92
+  SdfFilters sdf_filters = 6;
+  string app_id = 7;
+}
+
+//Group Type messages - Currently Includes -> 1. PDR 2. FAR
+
+//PDR message Table 7.5.2.2-1: Create PDR IE within PFCP Session Establishment Request
+message SetGroupPDR {
+  uint32 pdr_id = 1;
+  int32 sess_ver_no = 2;
+  uint32 precedence = 3;
+  PDI pdi = 4;
+  OuterHeadRemoval o_h_remo_desc = 5;      
+  uint32 far_id = 6;
+  uint32 urr_id = 7;
+  uint32 qer_id = 8;
+  uint32 bar_id = 9;
+  string active_pred_rule = 10;     // modified and renamed     //8.2.72 Doubt
+}
+
+
+//FAR message Table 7.5.2.3-1: Create FAR IE within PFCP Session Establishment Request
+message SetGroupFAR {
+  uint32 far_id = 1;              //FAR ID should be less than 2^32
+  int32 sess_ver_no = 2;
+  ApplyAction apply_action = 3;
+  FwdParam fwd_parm = 4;
+  DupParam du_param = 5;     // might be required for 4G
+  uint32 bar_id = 6;
+}
+
+// ToDO  SetGroupQER   write
+// ToDo  SetGroupBAR   write
+
+
+
+
+//SET message - SMF to Upf Session Requests
+message SessionSet {
+  string seid = 1;
+  int32 sess_ver_no = 2;
+  NodeID node_id = 3;
+
+  enum SessionState {
+    // before 6 all values are already reserved for old FSM states
+    CREATING = 7;
+    CREATED =  8;
+    ACTIVE   =9;
+    INACTIVE =10;
+    RELEASE  =11;
+  }
+  SessionState session_state = 4;
+  CpFSeid f_seid = 5;
+  repeated SetGroupPDR set_gr_pdr = 6;
+  repeated SetGroupFAR set_gr_far = 7;
+  //repeated SetGroupQER set_gr_qer = 8;
+  //repeated SetGroupBAR set_gr_bar = 9;
+  //TrafficEndPoint tr_ep = 9;
+  //PdnType pdn_type = 10;
+  //UserPlId user_pl_id = 11;
+  //UserId user_id = 12;
+  //TraceInfo tr_info = 13;
+  //ApnDnn apn_dnn = 14;
+}
+
+
+message UpfRes {}
+
```

Comment:
Remove void msgs.

Reply:
yes, it has been now updated and will be pushed soon,